lib/node_modules/@stdlib/_tools/changelog/generate/lib/npm_releases.js

```diff
@@ -0,0 +1,73 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var shell = require( 'child_process' ).execSync; // eslint-disable-line node/no-sync
+var logger = require( 'debug' );
+var omit = require( '@stdlib/utils/omit' );
+var rootDir = require( '@stdlib/_tools/utils/root-dir' );
+
+
+// VARIABLES //
+
+var debug = logger( 'changelog:generate' );
+var OMITTED_KEYS = [ 'created', 'modified', 'unpublished' ];
+
+
+// MAIN //
+
+/**
+* Returns a list of published package versions and their release dates.
+*
+* @private
+* @param {string} pkg - package name
+* @returns {Object} object mapping versions to release dates
+*
+* @example
+* var releases = npmReleases( '@stdlib/utils-omit' );
+* // returns {...}
+*/
+function npmReleases( pkg ) {
+	var command;
+	var opts;
+	var out;
+
+	command = 'npm view ' + pkg + ' time --json';
+	opts = {
+		'cwd': rootDir(),
+		'stdio': ['pipe', 'pipe', 'ignore'] // stdin, stdout, stderr
+	};
+	try {
+		out = shell( command, opts ).toString();
+		out = JSON.parse( out );
+		out = omit( out, OMITTED_KEYS );
```

Comment:
```suggestion
		out = shell( command, opts ).toString();
		out = JSON.parse( out );
		out = omit( out, OMITTED_KEYS );
```

We should move whatever expressions we do not except to error out of the `try/catch`; otherwise, you risk conflating syntax and operational errors.